core/templates/dev/head/pages/state_editor/StateSolutionEditorDirective.js

```diff
@@ -206,8 +208,13 @@ oppia.directive('stateSolutionEditor', [
                 StateEditorService.getActiveStateName(), solutionIsValid);
               $scope.refreshWarnings()();
               if (!solutionIsValid) {
-                AlertsService.addInfoMessage(
-                  INFO_MESSAGE_SOLUTION_IS_INVALID, 4000);
+                if (StateEditorService.isInQuestionMode()) {
+                  AlertsService.addInfoMessage(
+                    INFO_MESSAGE_SOLUTION_IS_INVALID_FOR_QUESTION, 4000);
+                } else {
+                  AlertsService.addInfoMessage(
+                    INFO_MESSAGE_SOLUTION_IS_INVALID, 4000);
```

Comment:
For parity, name this INFO_MESSAGE_SOLUTION_IS_INVALID_FOR_EXPLORATION

Reply:
Done.